src/summary/patient-chart-summary.component.tsx

```diff
@@ -10,44 +10,92 @@ import ProgramsCard from "./history/programs/programs-card.component";
 import MedicationsSummary from "./history/medications/medications-summary.component";
 
 export default function PatientChartSummary(props: PatientChartSummaryProps) {
-  const config = [
-    "conditions",
-    "programs",
-    "medications",
-    "allergies",
-    "notes",
-    "vitals",
-    "heightAndWeight"
+  const exampleConfig: configType[] = [
+    { name: "conditions" },
+    {
+      name: "Programs",
+      esModule: "@jj-widgets",
+      exportName: "programsWidget"
+    },
+    { name: "medications" },
+    { name: "allergies" },
+    { name: "notes" },
+    { name: "vitals" },
+    { name: "heightAndWeight" }
   ];
 
-  const coreComponents = {
-    conditions: ConditionsCard,
-    programs: ProgramsCard,
-    allergies: AllergyCard,
-    notes: NotesCard,
-    vitals: VitalsCard,
-    heightAndWeight: DimensionsCard,
-    medications: MedicationsSummary
-  };
+  const config: configType[] = [
+    { name: "conditions" },
+    { name: "medications" },
+    { name: "allergies" },
+    { name: "notes" },
+    { name: "vitals" },
+    { name: "heightAndWeight" },
+    { name: "programs" }
+  ];
+
+  const [widgets, setWidgets] = React.useState([]);
+
+  React.useEffect(() => {
+    const coreComponents = {
+      conditions: ConditionsCard,
+      programs: ProgramsCard,
+      allergies: AllergyCard,
+      notes: NotesCard,
+      vitals: VitalsCard,
+      heightAndWeight: DimensionsCard,
+      medications: MedicationsSummary
+    };
+
+    const modulePromises = [];
+
+    config.map(c => {
+      if (c["esModule"]) {
+        modulePromises.push(System.import(c.esModule));
+      }
+    });
```

Comment:
Let's remember to refactor this so that these promises only happen once. Probably just in the root component, and then the router passes the widgets to the pages.